File: max-consecutive-ones.py
from typing import List


def findMaxConsecutiveOnes(nums: List[int]) -> int:
    if not nums:
        return 0
    elif nums == [0]:
        return 0
    elif nums == [1]:
        return 1
    elif set(nums) == {0}:
        return 0
    elif set(nums) == {1}:
        return len(nums)
    else:
        current_total: int = 0
        current_max: int = 0

        for num in nums:
            if num == 1:
                current_max += 1
            else:
                current_total = max(current_total, current_max)
                current_max = 0

        return max(current_total, current_max)


findMaxConsecutiveOnes([1, 1, 0, 1, 1, 1])

File: java/src/main/java/dev/bradcodecraft/app/MaxConsecutiveOnes.java
package dev.bradcodecraft.app;

/*
 * Given a binary array nums, return the maximum number of consecutive 1's in the array.
 */
public class MaxConsecutiveOnes {
  public int findMaxConsecutiveOnes(int[] nums) {
    if (nums.length == 0 || nums.equals(new int[] { 0 })) {
      return 0;
    } else if (nums.equals(new int[] { 1 })) {
      return 1;
    } else {
      int currentTotal = 0;
      int currentMax = 0;

      for (int number : nums) {
        if (number == 0) {
          currentTotal = Math.max(currentMax, currentTotal);
          currentMax = 0;
        } else {
          currentMax++;
        }
      }

      return Math.max(currentMax, currentTotal);
    }
  }
}
